### PR TITLE
fix(replays): fix types on replays.tasks

### PR DESCRIPTION
**Pull request by JoshFerge**

### CODE REVIEW

File: src/sentry/replays/tasks.py [hunk 1]
```diff
@@ -28,22 +31,23 @@ def delete_recording_segments(project_id: int, replay_id: str, **kwargs: dict) -
...
-    for segment in segment_sql_rows:
-        segment.delete()  # Three queries + one request to the message broker
+    for segment_model in segments_from_django_models:
+        segment_model.delete()  # Three queries + one request to the message broker
 
 
 def archive_replay(project_id: int, replay_id: str) -> None:
     """Archive a Replay instance. The Replay is not deleted."""
-    replay_payload = {
+    replay_payload: dict[str, Any] = {
```
Comment (untitaker) [inline src/sentry/replays/tasks.py:50]: https://github.com/getsentry/sentry-kafka-schemas/pull/143

:)

**Review by untitaker — APPROVED** [review]

File: src/sentry/replays/tasks.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/replays/tasks.py:50]: weird, i get 
```
src/sentry/replays/tasks.py:50: error: Need type annotation for "replay_payload"  [var-annotated]
```
when removed

Comment (untitaker) [reply src/sentry/replays/tasks.py:50]: yeah just saying you could have better types here that are actually enforced across services (incl relay)

Comment (JoshFerge) [reply src/sentry/replays/tasks.py:50]: (deleted my comment, didn't see your link haha). yeah 100%. we should def. work towards that.

**Pull request merged**